### PR TITLE
Upgrades `mypy` to version `0.971`

### PR DESCRIPTION
**Pull request by dmitryduev**
Fixes WB-NNNN
Fixes #NNNN

Description
-----------
- Upgrade `mypy` to 0.971.
- Remove module-level imports of `parse_version` from `pkg_resources`, which takes way too much time.

Testing
-------
How was this PR tested?

Checklist
-------
- Include reference to internal ticket "Fixes WB-NNNN" (and github issue "Fixes #NNNN" if applicable)

### CODE REVIEW

**Review by kptkin — APPROVED** [review]

File: wandb/sdk/internal/stats.py [hunk 1]
```diff
@@ -150,7 +150,7 @@ def _thread_body(self) -> None:
                 time.sleep(0.1)
                 seconds += 0.1
                 if self._shutdown:
-                    self.flush()
+                    self.flush()  # type: ignore
```
Comment (raubitsj) [inline wandb/sdk/internal/stats.py:153]: is this expected that you need a type ignore?

File: wandb/sdk/wandb_run.py [hunk 2]
```diff
@@ -3199,20 +3219,23 @@ def _footer_multiple_runs_file_pusher_status_info(
                 ]
             )
             for response in poll_exit_responses
-            if response and response.file_counts
+            if response is not None and response.file_counts is not None
```
Comment (raubitsj) [inline wandb/sdk/wandb_run.py:3222]: is this your style or is this mypy enforcing this?

**Review by raubitsj — APPROVED** [review]
great!

File: wandb/sdk/wandb_run.py (hunk 2 above)
Comment (dmitryduev) [reply wandb/sdk/wandb_run.py:3222]: mypy!

File: wandb/sdk/internal/stats.py (hunk 1 above)
Comment (dmitryduev) [reply wandb/sdk/internal/stats.py:153]: yes, it's calling not-fully-typed functions. i might get to ++'ing our annotation coverage later in the week/next week, will fix then. my main task here was to make it pass on the latest mypy version. :)

**Pull request merged**